product-scenarios/10-file-processing/10.1-simple-file-movement-between-two-systems/10.1.1-reading-a-file-from-a-specified-location-in-file-system/10.1.1.7-reading-a-file-from-local-location/src/test/java/org/wso2/carbon/esb/scenario/test/ReadFileFromLocalLocationTest.java

```diff
@@ -0,0 +1,146 @@
+import org.apache.http.HttpResponse;
+import org.awaitility.Awaitility;
+import org.json.JSONObject;
+import org.testng.Assert;
+import org.testng.annotations.BeforeClass;
+import org.testng.annotations.Test;
+import org.wso2.carbon.esb.scenario.test.common.ScenarioConstants;
+import org.wso2.carbon.esb.scenario.test.common.ScenarioTestBase;
+import org.wso2.carbon.esb.scenario.test.common.StringUtil;
+import org.wso2.carbon.esb.scenario.test.common.http.RESTClient;
+
+import java.io.IOException;
+import java.util.Map;
+import java.util.concurrent.Callable;
+import java.util.concurrent.TimeUnit;
+
+import static org.wso2.carbon.esb.scenario.test.common.http.HTTPUtils.getResponsePayload;
+
+/**
+ * This test class test reading a file from a specified location in file system
+ */
+public class ReadFileFromLocalLocationTest extends ScenarioTestBase {
+
+    private static final Log log = LogFactory.getLog(ReadFileFromLocalLocationTest.class);
+    private static String queryFilePath = "&filePath=";
+    private static String querySourcePath = "&sourcePath=";
+    private static String queryDestinationPath = "&destinationPath=";
+    private static String queryFileName = "&fileName=";
+
+    @BeforeClass(description = "Test init")
+    public void init() throws Exception {
+        super.init();
+    }
+
+    /**
+     * This test case test reading a file from a local location using VFS listener
+     *
+     * @throws IOException - if any REST API request fails
+     */
+    @Test(description = "10.1.1.7.1")
+    public void testReadFileFromLocalLocationUsingVFSListener() throws IOException {
+        String messageID = "10.1.1.7.1";
+        String createInputDirUrl = getApiInvocationURLHttp("fileOperation") + "/createFolder?" +
+                                   queryFilePath + getLocalVfsLocation("vfs/input");
+
+        String createOutputDirUrl = getApiInvocationURLHttp("fileOperation") + "/createFolder?" +
+                                    queryFilePath + getLocalVfsLocation("vfs/output");
+
+        String createOriginalDirUrl = getApiInvocationURLHttp("fileOperation") + "/createFolder?" +
+                                      queryFilePath +  getLocalVfsLocation("vfs/original");
+
+        String quoteRequestDirUrl = getVfsSourceDir("vfs/source_files");
+
+        String copyQuoteRequestUrl = getApiInvocationURLHttp("fileOperation") + "/copyFile?" +
+                                     querySourcePath + quoteRequestDirUrl +
+                                     queryDestinationPath + getLocalVfsLocation("vfs/input") +
+                                     queryFileName + "requestQuote.xml";
+
+        String stockQuoteFileExistUrl = getApiInvocationURLHttp("fileOperation") + "/isFileExist?" +
+                                        queryFilePath + getLocalVfsLocation("vfs/output") + "/stockQuote.xml";
+
+        doPutRestClient(createInputDirUrl, null, "", "application/json", "Create input directory");
```

Comment:
We have a problem here since we have a LB fronting EI cluster we cannot guarantee in which node directory get created

Reply:
Let's disable this test till we find solution